### PR TITLE
Integrate with Cypress for e2e testing

### PR DESCRIPTION
**Pull request by Linco1n3**

### CODE REVIEW

File: src/test/e2e-testing-cypress/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Cypress E2E Testing
+
+Cypress is a next generation front end testing tool built for the modern web. Cypress is a complete end to end testing framework that runs in the browser. 
+
+## Installation
+
+```shell
+npm install --save-dev cypress
```
Comment (weihanwang) [inline src/test/e2e-testing-cypress/README.md:8]: don't use `npm`. Use `yarn install` instead.

File: yarn.lock
Comment (weihanwang) [inline yarn.lock:1]: restore this file first and use `yarn install` to update it.

Comment (Linco1n3) [reply yarn.lock:1]: done

File: src/test/e2e-testing-cypress/README.md (hunk 1 above)
Comment (Linco1n3) [reply src/test/e2e-testing-cypress/README.md]: done

File: src/test/e2e-testing-cypress/README.md
[the remark this answers is not in the document]
Comment (Linco1n3) [reply src/test/e2e-testing-cypress/README.md:1]: done

**Review by weihanwang — APPROVED** [review]

**Pull request merged**